tests/ValidatorDelegates/CSPBadFlagsTest.php

```diff
@@ -0,0 +1,182 @@
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "default-src 'unsafe-inline' 'unsafe-eval'; "
+                . "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideBadFlagTestCases
+     * @param $headerValue
+     * @param $expectedErrors
+     */
+    public function testBadFlags($headerValue, $expectedErrors)
+    {
+        $BadCSP = HeaderFactory::build('Content-Security-Policy', $headerValue);
+
+        $Errors = CSPBadFlags::validate($BadCSP);
+
+        if ([] === $expectedErrors)
+        {
+            $this->assertCount(0, $Errors);
+        }
+        else
+        {
+            $this->assertEquals($expectedErrors, $Errors);
```

Comment:
I'm not sure `assertEquals` is sufficient to test arrays of `Error`, other than that their counts are equal.  This is the code that generates the string being compared for each element:  https://github.com/sebastianbergmann/exporter/blob/master/src/Exporter.php#L122  So that will be doing something like
```
if ("Error Object (n)" !== "Error Object (n)") throw("unequal");
```

So this might need a manual cycle through the array checking the `toString` of the object.